client/live_test.go

```diff
@@ -62,3 +62,14 @@ func (s *LiveTests) TestAuthenticate(c *gc.C) {
 	c.Check(err, gc.IsNil)
 	c.Check(url, gc.NotNil)
 }
+
+func (s *LiveTests) TestAuthDiscover(c *gc.C) {
+	if s.authMode == identity.AuthLegacy {
+		c.Skip("this test will not work with legacy auth")
+	}
+	cl := client.NewClient(s.cred, s.authMode, nil)
+	options, err := cl.IdentityAuthOptions()
+	c.Assert(err, gc.IsNil)
+	optionsAvailable := len(options) > 0
```

Comment:
please test that the options available values belong to the set of the ones we expect